builtin/rebase--helper.c

```diff
@@ -97,9 +144,9 @@ int cmd_rebase__helper(int argc, const char **argv, const char *prefix)
 		return !!edit_todo_list(flags);
 	if (command == PREPARE_BRANCH && argc == 2)
 		return !!prepare_branch_to_be_rebased(&opts, argv[1]);
-	if (command == COMPLETE_ACTION && argc == 6)
-		return !!complete_action(&opts, flags, argv[1], argv[2], argv[3],
-					 argv[4], argv[5], autosquash, keep_empty);
+	if (command == COMPLETE_ACTION && argc == 3)
+		return !!complete_action(&opts, flags, shortrevisions.buf, argv[1], onto,
+					 head_hash, argv[2], autosquash, keep_empty);
```

Comment:
So the short revisions are only needed here? The we also should only generate it here. You can bring this about by teaching aforementioned function to handle `shortrevisions != NULL` and only doing the "short dance" in that case.